classes/02class/exercises/c02-network03/README.md

```diff
@@ -0,0 +1,146 @@
+|-|-|
+|0.0.0.0|`devopsacademy-igw`|
+
+- devopsacademy-rt-private
+
+|destination|target|
+|-|-|
+|0.0.0.0|`devopsacademy-ngw`|
+
+#### Then post your answers for the following:
+
+- How did you accomplish both instructions?
+- Why did you add 0.0.0.0 route to the IGW on public route table?
+- Why did you add 0.0.0.0 route to the NGW on private route table?
+- What is the difference of IGW and NGW?
+- Can you delete the destination route to your VPC network? Why?
+- What happens if you do not associate your route table with any subnets?
+
+### Network ACL (NACL's)
+
+#### Look for the network ACL created for your VPC and answer the following:
+
+Your ACL rule inbound and outbound should look like this:
+
+|Rule#|Type|Protocol|Port Range|Source|Allow/Deny|
+|-|-|-|-|-|-|
+|100|ALL Traffic|ALL|ALL|`<vpc-network>`|ALLOW|
+|101|ALL Traffic|ALL|ALL|0.0.0.0/0|ALLOW|
+|*|ALL Traffic|ALL|ALL|0.0.0.0/0|DENY
+
+- What is the difference between inbound x outbound?
+- What is going to happen by default? Everything allowed or denied?
+- How would you block SSH connection comming from the network `172.16.200.200/24`?
+- How would you block access from your network to an external network `180.200.50.0/24`?
+
+### Security Group (SG)
+
+Create two new security groups with attached to your new VPC with the following:
+
+|Security Group Name|Rule|Type|Port Range|Source|Description
+|-|-|-|-|-|-|
+|public-sg|inbound|Custom TCP Rule|22|`<your-home-ip-address>`|SSH access from home to AWS|
+|private-sg|inbound|Custom TCP Rule|22|`<public-sg-id>`|SSH access only from public instances on AWS|
+
+#### post details on how did you achieve that
+
+### Elastic Network Interface (ENI)
+
+Let's assume our service will need a fixed IP address and that if something happens with the instance we will have to move this IP address to a new one.
+
+Create a new ENI attached to one of your private subnets and give it a fixed IP address within the network range of the chosen subnet.
+
+#### post details on how did you achieve that -- note that Network Interfaces can be found on Services > Computer > EC2
+
+### EC2
+
+Repeat exercise from last class where you have to:
+
+- launch an EC2 instance on a public subnet
+  - attach your public SG to your public instance
+- launch an EC2 instance on the same private subnet that you created your ENI
+  - attach ENI created from last exercise
+  - attach your private SG to your public instance
+- access to your public instance using SSH
+- FROM your PUBLIC instance, SSH to your private one
+
+However this time, use your custom created subnets instead of the AWS default ones.
+
+#### post if you had to do any troubleshooting or if that worked straight away
+
+
+#### DO NOT FORGET TO DELETE YOUR RESOURCES ONCE YOU FINISH:
+
+- ALL EC2 INSTANCES
+- NAT GATEWAY
+- ANY ELASTIC IP (those are charged only if you allocate one and is not being used)
+
+You can leave the vpc, subnets, route tables, internet gateway, network acl
+
+
+
```

Comment:
as a whole, I think we need to provide instructions on what `Post how did you accomplish that on AWS console.`, and similar requests, means ?
Something like:
Create a PR with the following files:
1) ex01.txt, containing:
a) xxxxx?
b) yyyyy?
2) ex02.txt, containing:
a) wwww?
b) zzzzz?
Also, it seems to be a huge single exercise, so I'd  suggest to split it in multiple files.

Reply:
I also thought that it's a big exercise, but it's pointless without the whole thing

Reply:
If you think that a single file all the answers is the best way to go for this, let's move on this way. I just think that this will make people hold their PR until they have everything is place, and that might take a long time. there won't be any small  accomplishment, just a big one.

Reply:
split into multiple files, let's see how it goes